pymeasure/instruments/agilent/agilent4156.py

```diff
@@ -0,0 +1,1052 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2017 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import (Instrument,
+                                   RangeException)
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              truncated_discrete_set,
+                                              strict_range)
+import numpy as np
+import time
+import pandas as pd
+import os
+import json
+
+######
+# MAIN
+######
+
+
+class Agilent4156(Instrument):
+    """ Represents the Agilent 4155/4156 Semiconductor Parameter Analyzer
+    and provides a high-level interface for taking current-voltage (I-V) measurements.
+
+    .. code-block:: python
+
+        from pymeasure.instruments.agilent.agilent4156 import Agilent4156
```

Comment:
You can actually import this class directly from `pymeasure.instruments.agilent` because of the `__init__.py` line that you added.